passacre/config.py

```diff
@@ -303,10 +307,37 @@ def set_config(self, site, name, value):
                 (site, name, jdumps(new_value)))
         self._db.commit()
 
+class SqliteTahoeConfig(SqliteConfig):
+    @features.tahoe.check
+    def __init__(self):
+        try:
+            with open("~/.tahoe/node.url") as f:
+                self.tahoe = f.readline().strip()
+                self.tahoe += "uri/" if self.tahoe.endswith("/") else "/uri/"
+        except IOError:
+            self.tahoe = "http://localhost:3456/uri/"
+        super(SqliteConfig, self).__init__()
+
+    def read(self, uri):
+        "Load site configuration from Tahoe-LAFS gateway"
+        import requests
+        import sqlite3
+        self._db_uri = "{}{}".format(self.tahoe, uri.readline().strip())
+        self._db_raw = NamedTemporaryFile(prefix='passacre')
+        db_request = requests.get(self._db_uri)
+        db_request.raise_for_status()
+        self._db_raw.write(db_request.content)
+        self._db_raw.file.flush()
+        self._db_raw.file.seek(0)
+        super(SqliteTahoeConfig, self).read(self._db_raw)
+
 
 def load(infile):
-    if infile.read(16) == b'SQLite format 3\x00':
+    magic = infile.read(16)
+    if magic == b'SQLite format 3\x00':
         config = SqliteConfig()
+    elif magic.startswith('URI:MDMF:') or magic.startswith('URI:SSK:'):
```

Comment:
These need to be `bytes` literals for Python 3. `startswith` also accepts a tuple.